### PR TITLE
[Refactor:API] Migration for Semester to Term

### PR DESCRIPTION
**Pull request by NiaHeermance**
### What is the current behavior?
Currently, in the code, we use the word "term" and "semester" interchangeably. Our goal is to shift just to using "term" to be less confusing for schools without a semester system.

### What is the new behavior?
The database is now free of "semester." "semester" still appears in the code but no longer in the database.

### How to test:
Navigate to various pages on Submitty and see if anything breaks. I've successfully vagrant up'd from scratch on this branch (as well as of course migrated to it from a vagrant up on ``main``), and I've done a lot of testing so it should be good, but more testing is a great thing.

### CODE REVIEW

**Review by 04vmatsibekker — CHANGES_REQUESTED** [review]
image

The calendar uses stuff involving semester and course name. When I tried to upload an announcement as instructor for the sample course, I got this error message. I believe it might have something to do with the changes you made in your PR.

**Comment by NiaHeermance** [comment]
> image
> 
> The calendar uses stuff involving semester and course name. When I tried to upload an announcement as instructor for the sample course, I got this error message. I believe it might have something to do with the changes you made in your PR.

Fixed!

**Review by pbailie — CHANGES_REQUESTED** [review]
Looks like you missed a `semester` token somewhere.  After running the DB migration in Vagrant, I saw the following exception thrown:
```
Oh no! Something irrecoverable has happened...

app\exceptions\DatabaseException (Code: 0) thrown in /usr/local/submitty/site/app/libraries/database/AbstractDatabase.php (Line 182) by:
            throw new DatabaseException($pdoException->getMessage(), $query, $parameters);


Message:
SQLSTATE[42703]: Undefined column: 7 ERROR:  column u.semester does not exist
LINE 3: INNER JOIN courses c ON u.course=c.course AND u.semester=c.s...
                                                      ^

Stack Trace:
#0 /usr/local/submitty/site/app/libraries/database/DatabaseQueries.php(4597): app\libraries\database\AbstractDatabase->query()
#1 /usr/local/submitty/site/app/controllers/HomePageController.php(48): app\libraries\database\DatabaseQueries->getCourseForUserId()
#2 /usr/local/submitty/site/app/controllers/HomePageController.php(104): app\controllers\HomePageController->getCourses()
#3 unknown file(unknown line): app\controllers\HomePageController->showHomepage()
#4 /usr/local/submitty/site/app/libraries/routers/WebRouter.php(262): call_user_func_array()
#5 /usr/local/submitty/site/app/libraries/routers/WebRouter.php(240): app\libraries\routers\WebRouter->run()
#6 /usr/local/submitty/site/public/index.php(131): app\libraries\routers\WebRouter::getWebResponse()
Extra Details:
query: SELECT t.name AS term_name, u.semester, u.course, u.user_group, u.registration_section
FROM courses_users u
INNER JOIN courses c ON u.course=c.course AND u.semester=c.semester
INNER JOIN terms t ON u.semester=t.term_id
WHERE u.user_id=? AND c.status=1 AND (u.registration_section IS NOT NULL OR u.user_group<>4)
ORDER BY u.user_group ASC, t.start_date DESC, u.course ASC
parameters:
instructor
```

**Review by pbailie — COMMENTED** [review]
My primary concern is compatibility with sysadmin scripts (student autofeed and account creation).  Changes in the sysadmin scripts are required to work with this PR.

I did a *__cursory__* look in Vagrant, and after migrating both system and course DBs —  things appear to be OK.  Due to the enormous scale (and risk) of this refactor, another review is advised.

File: site/app/libraries/database/DatabaseQueries.php [hunk 1]
```diff
@@ -1159,7 +1159,7 @@ public function updateGradingRegistration($user_id, $user_group, $sections) {
      * @return integer - group number of user in the given class
      */
     public function getGroupForUserInClass($semester, $course_name, $user_id) {
```
Comment (ziesski) [inline site/app/libraries/database/DatabaseQueries.php:1161]: I noticed that a lot of variable are still named $semester, will these be changed in another round of refactor?

Comment (NiaHeermance) [reply site/app/libraries/database/DatabaseQueries.php:1161]: Yep!

**Comment by NiaHeermance** [comment]
> My primary concern is compatibility with sysadmin scripts (student autofeed and account creation). Changes in the sysadmin scripts are required to work with this PR.
> 
> I did a _**cursory**_ look in Vagrant, and after migrating both system and course DBs — things appear to be OK. Due to the enormous scale (and risk) of this refactor, another review is advised.

I actually edited the sysadmin scripts in this PR: https://github.com/Submitty/SysadminTools/pull/33.

**Pull request merged**